python/ray/train/callbacks/callback.py

```diff
@@ -16,21 +16,27 @@ def handle_result(self, results: List[Dict], **info):
         """
         pass
 
-    def start_training(self, logdir: str, **info):
+    def start_training(self, logdir: str, config: Dict, **info):
         """Called once on training start.
 
         Args:
             logdir (str): Path to the file directory where logs
                 should be persisted.
+            config (Dict): The config dict passed into ``trainer.run()``.
             **info: kwargs dict for forward compatibility.
         """
         pass
 
-    def finish_training(self, error: bool = False, **info):
+    def finish_training(self,
+                        error: bool = False,
+                        run_dir: Optional[str] = None,
+                        **info):
         """Called once after training is over.
 
         Args:
             error (bool): If True, there was an exception during training.
+            run_dir (Optional[str]): The path to the directory for this
+                training run.
```

Comment:
Also for consistency with `start_training` should this be called `logdir`? Is this meant to be the same directory? Any way this could just be passed in once to avoid inconsistencies?

Reply:
Related: Shouldn't the callbacks just accept a `trainer: Trainer` argument?

Reply:
@matthewdeng you're right this argument probably isn't necessary since we can access the `logdir` anyways via `self.logdir`. We don't need to pass it in again in `finish_training`.

Reply:
@krfricke I think for now we want to abstract way the trainer from the callbacks and only pass in the information that's needed. I agree, we should reconsider the interfaces here, but it's out of scope for this PR.